tokio-codec/src/lines_codec.rs

```diff
@@ -41,15 +105,49 @@ impl Decoder for LinesCodec {
     type Error = io::Error;
 
     fn decode(&mut self, buf: &mut BytesMut) -> Result<Option<String>, io::Error> {
-        if let Some(newline_offset) =
-            buf[self.next_index..].iter().position(|b| *b == b'\n')
-        {
-            let newline_index = newline_offset + self.next_index;
+        let mut trim_and_offset = None;
+        for (offset, b) in buf[self.next_index..].iter().enumerate() {
+            trim_and_offset = match (b, self.max_length) {
+                // The current character is a newline, split here.
+                (&b'\n', _) => Some((1, offset)),
+                // There's a maximum line length set, and we've reached it.
+                (_, Some(max_len))
+                    if offset.saturating_sub(self.next_index) == max_len => {
+                    // If we're at the line length limit, check if the next
+                    // character(s) is a newline before we decide to return an
+                    // error.
+                    match &buf[offset + 1..=offset + 2] {
```

Comment:
Errant equals sign or have I just really not been paying attention to the Rust docs? 😆

Reply:
The equals sign [indicates an inclusive range](https://doc.rust-lang.org/std/ops/struct.RangeInclusive.html). However, that syntax was added in Rust 1.26.0, so I removed it in 2e1076d8dd8ad36b124d575e56082509040f054d to fix the Rust 1.25 CI build.